### ISSUE TITLE
TIM DMA transfer example required for better understanding

### ISSUE THREAD

**Issue by Slamy** [stm32-rs/stm32f4xx-hal#530]
I realize that this was also discussed in #227 but the issue still seems to persist even so the ticket was closed.
I was also curious how this particular person has solved it to avoid asking here again.
I've found this code snippet in https://github.com/justdimaa/droners/blob/main/src/main.rs:
`
let ccr1_tim3 =
            dma::traits::CCR1::<pac::TIM3>(unsafe { mem::transmute_copy(&device.TIM3) });
`
As I don't assume that this is the right way to provide registers to the DMA Transfer I just have to reopen this issue here again.
This project lacks an example which uses a timer with the DMA. Either with input capture or output compare.
Maybe I'm still too inexperienced with embedded Rust but after hours of trying I gave up.

**Comment by burrbull** [comment]
> Maybe I'm still too inexperienced with embedded Rust but after hours of trying I gave up.

You are absolutely right. There is not abstraction for timer/pwm driven by DMA in this HAL.
Although it should be not very hard to write it as most of building blocks are already there.
